axelrod/fingerprint.py

```diff
@@ -65,168 +28,91 @@ def __init__(self, strategy, probe=axl.TitForTat):
             A class that must be descended from axelrod.Player or an instance of
             axelrod.Player.
             Default: Tit For Tat
+        step: float = 0.01
+            The separation between each Point. Smaller steps will
+            produce more Points that will be closer together.
+        progress_bar: bool = True
+            True = show a progress bar when constructing probes and running
+            tournament.
         """
-        self.strategy = strategy
-        self.probe = probe
+        self.player = create_player(strategy)
+        self._probe_class, self._probe_kwargs = get_class_and_kwargs(probe)
+        self._step = step
+        self.progress_bar = progress_bar
+        self.points = self._create_points()
+        self._probe_list = self._create_probes()
```

Comment:
Yup, there's nothing wrong with that thought process. Another way of thinking about it though is that a fingerprint is an abstract thing (technically it's an analytical function) and to `fingerprint` is to create a tournament (ie probes which is also why the `turns, steps` etc are in there) and play the tournament.

Given that the status quo is already implemented I suggest it stays put (otherwise someone else might come along at another point with a different way of thinking about it and change it to that etc...). If there's a reason other than your interpretation that I'm missing though that's worth discussing.

The other argument in favour of the status quo is that `.fingerprint()` is where all the work happens (for a small step size, creating probes can actually take a long time). This is similar to the rest of the library where `Match.play()`, `Tournament.play()` and `Moran.play()` does all the work.